tests/unit/mixins/adapter-fetch-test.js

```diff
@@ -348,6 +359,47 @@ test('determineBodyResponse returns an empty object when the request method is \
   });
 });
 
+test('parseFetchResponseForError is able to be overwritten to mutate the error payload that gets passed along', function(assert) {
+  assert.expect(1);
+
+  this.errorAdapter = JSONAPIAdapter.extend(AdapterFetchMixin, {
+    _fetchRequest() {
+      const response = new Response(`{ "errors": ["myoneerror"] }`, {status: 422});
+      return Ember.RSVP.Promise.resolve(response);
+    },
+    parseFetchResponseForError() {
+      return {
+        errors: ['myOverWrittenError']
+      }
+    }
+  }).create();
+
+  const fetchReturn = this.errorAdapter.ajax({url: '/foo'});
```

Comment:
to remove confusion, can you change this to `ajax('/foo')` to match ajax's signature in this mixin